### PR TITLE
Typography and little bits

### PR DESCRIPTION
**Pull request by dandwal**
#### Typography

- Open Sans font added for regular and bold. It will attempt to load locally and then from the CDN in this order: `eot`, `woff2`, `woff`, `truetype`, `svg`
- Headings, default styling added for h1 to h4

#### Hide

- Like in sixteens, 'dp-hidden' will hide an element from all media
- `dp-visuallyhidden` will hide elements visually but not from screenreaders
- `dp-hide` will hide an element for a given media type based on screen size
dp-hide example
```scss
<h1 dp-hide--for-xxs>Can only see me on screens bigger than the xsss breakpoint</h1>
<h2 dp-hide--for-s-up>Can only see me on screens smaller than the s breakpoint, xs and xss</h2>

```

#### Breakpoint

Based on the universal system in the 'design-system'
- "xxs": 300px
- "xs": 400px
- "s": 500px
- "m": 740px
- "l": 980px
- "xl": 1300px
- "xxl": 1600px

#### Generic styling

- All padding and margin removed, margin and padding should be applied as needed by design
- Borders now included as part of the width calculations of an element
- HTML and body should take up the full viewport
 
### How to review

Take this out for a spin, check that it conforms to any standards decided

### Who can review

Anyone except me

### CODE REVIEW

**Review by hamishtaplin — APPROVED** [review]
LGTM 👍

File: scss/abstracts/variables/_typography.scss [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                weight: $font-weight-bold,
+                line-height: 1.4
+        ),
+        dp-fs-r--b: (
+                small: $base,
+                large: $base,
+                weight: $font-weight-bold,
+                line-height: 1.4
+        ),
+        dp-fs-r: (
```
Comment (rav-pradhan) [inline scss/abstracts/variables/_typography.scss:46]: Just to clarify, is the `r` to signify `regular`?

**Review by rav-pradhan — COMMENTED** [review]
Tried out some of the styling in a separate project and looks fine. I've left some comments, the main one related to overriding system/browser set default font sizes in the html tag (open for opinions on it)

File: scss/base/_typography.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
+html {
+  font-size: $base; /* set 1rem to value of $base */
```
[the remark this answers is not in the document]
Comment (dandwal) [reply scss/base/_typography.scss:2]: Hi, yer I actually raised this exact issue in one of the working group meetings :D, UX weighed in and it was agreed that we should take this route. But might be worth throwing the question into the slack channel to verify?

File: scss/abstracts/functions/_conversions.scss [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
+@function pxToRems($px, $base: $base) {
```
[the remark this answers is not in the document]
Comment (dandwal) [reply scss/abstracts/functions/_conversions.scss:2]: Ah, can't take credit stole from the DST design system :D

File: scss/abstracts/variables/_typography.scss (hunk 1 above)
Comment (dandwal) [reply scss/abstracts/variables/_typography.scss:46]: Yer you got it, I think in the design guide maybe we can have references to what the shorthand initials mean, or if preferred I could add a comment in code, to save people having to look around.

File: scss/base/_typography.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (rav-pradhan) [reply scss/base/_typography.scss:2]: Yeah probably will do just to better understand the rationale, but if it's been agreed and justified with UX then I'm happy to approve as well for now 👍🏽

File: scss/abstracts/variables/_typography.scss (hunk 1 above)
Comment (rav-pradhan) [reply scss/abstracts/variables/_typography.scss:46]: Yeah, that's a good shout 😃

**Review by rav-pradhan — APPROVED** [review]

**Pull request merged**